medusa/providers/torrent/json/eztv.py

```diff
@@ -0,0 +1,150 @@
+# coding=utf-8
+
+"""Provider code for EZTV."""
+
+from __future__ import unicode_literals
+
+import logging
+import time
+
+from medusa import tv
+from medusa.helper.common import convert_size
+from medusa.indexers.utils import mappings
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class EztvProvider(TorrentProvider):
+    """EZTV Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(EztvProvider, self).__init__('Eztv')
+
+        # Credentials
+        self.public = True
+
+        # URLs
+        self.url = 'https://eztv.io'
+        self.urls = {
+            'api': 'https://eztv.io/api/get-torrents',
+        }
+
+        # Proper Strings
+
+        # Miscellaneous Options
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=15)
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+
+        # Search Params
+        search_params = {
+            'imdb_id': '*',
+            # 'limit': 30,
+        }
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+
+                if mode != 'RSS':
+                    imdb_id = self.series.externals.get(mappings[10])
+                    imdb_id = imdb_id[2:]  # strip two tt's of id as they are not used by eztv

```

Comment:
Please put this after the if check below, as there are shows that don't have an imdb entry and it would cause an exception otherwise

Reply:
Atually this can not be moved. Every show on eztv works based on imdb. So if a show has no IMDB the show will not be on EZTV. Therefor the else of the if statement shows log.debug('IMDb ID not found'). Please verify i do not have any shows without IMDB.

Reply:
Shows without IMDB will not be able to be search through the API. This how their API works see https://eztv.io/api/

Reply:
I belive what he was saying is to move the slicing to right after the imdb_id check, so after line 68

Reply:
Yeah because if returns null then errors out if sliced. makes sense!

Reply:
Just pushed this in latest commit but travis is failing me 😕